File: application/src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/types/GroupByTypes.java
package de.pbauerochse.worklogviewer.youtrack.v20174.types;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class GroupByTypes implements GroupingField {

    private final String id;
    private final String presentation;

    @JsonCreator
    public GroupByTypes(@JsonProperty("id") String id,
                        @JsonProperty("presentation") String presentation) {
        this.id = id;
        this.presentation = presentation;
    }

    @Override
    public String getId() {
        return id;
    }

    @Override
    public String getPresentation() {
        return presentation;
    }
}

File: connectors/v20182/src/main/java/de/pbauerochse/worklogviewer/connector/v2018/domain/groupby/GroupByTypes.kt
package de.pbauerochse.worklogviewer.connector.v2018.domain.groupby

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
class GroupByTypes @JsonCreator constructor(
    @JsonProperty("id") override val id: String,
    @JsonProperty("presentation") override val presentation: String
) : GroupingField {

    override fun getPossibleNames(): Iterable<String> = arrayListOf(id, presentation)

}